codegen/smithy-go-codegen-test/model/main.smithy

```diff
@@ -7,8 +7,10 @@ use smithy.waiters#waitable
 
 /// Provides weather forecasts.
 @httpBearerAuth
+@httpApiKeyAuth(scheme: "Apikey", name: "Authorization", in: "header")
 @fakeProtocol
 @paginated(inputToken: "nextToken", outputToken: "nextToken", pageSize: "pageSize")
+@auth([httpBearerAuth, httpApiKeyAuth])
```

Comment:
general note: `@auth` order is not implemented in smithy-go yet. When testing locally, changing the order as shown produced the same order of auth middleware resolution as defined in `AddAuthPlugin.java`:

```smithy
@auth([httpApiKeyAuth, httpBearerAuth])
```